torchvision/models/detection/rpn.py

```diff
@@ -163,6 +163,8 @@ def forward(self, image_list, feature_maps):
                 anchors_in_image.append(anchors_per_feature_map)
             anchors.append(anchors_in_image)
         anchors = [torch.cat(anchors_per_image) for anchors_per_image in anchors]
+        # Clear the cache in case that memory leaks.
+        self._cache.clear()
```

Comment:
I think this is a bit too drastic: the size of the cache is in general fairly small (less than 1MB per key), and we can get some speedup if we cache the most used sizes. This is effectively removing the caching altogether.

Maybe for a follow-up PR: What about using some variant of [`functools.lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache) instead of manually keeping a dictionary for the cache? 

This way, we could potentially keep save the 32 last sizes.